docs/dg/dev/guidelines/performance-guidelines/infrastructure-worker-configuration-guidelines.md

```diff
@@ -0,0 +1,347 @@
+```
+
+**2. Asynchronous loading:**
+
+For third-party scripts and widgets, load them asynchronously in the browser:
+
+```html
+<!-- Third-party analytics loads directly from provider -->
+<script async src="https://analytics-provider.com/tracker.js"></script>
+```
+
+**When proxying might be necessary:**
+
+- **Security requirements**: Hiding internal services behind reverse proxy
+- **Legacy integrations**: Systems that require single origin
+- **Authentication injection**: Adding auth headers to external requests
+
+Even in these cases, proxy only specific, necessary endpoints, not entire external services.
+
+### Adjust nginx and PHP-FPM buffers for response size
+
+nginx and PHP-FPM use buffers to handle request/response data. Default buffer sizes (4-8KB) work for small responses but cause performance issues with larger payloads common in e-commerce applications.
+
+### Understanding the buffer problem
+
+**How nginx + PHP-FPM communicate:**
+
+1. nginx receives HTTP request from client
+2. nginx forwards request to PHP-FPM via FastCGI
+3. PHP-FPM processes request and generates response
+4. PHP-FPM sends response back through FastCGI buffers
+5. nginx receives buffered response and sends to client
+
+**What happens with undersized buffers:**
+
+When response size exceeds buffer capacity:
+- PHP-FPM must write to temporary files on disk (slow I/O)
+- nginx reads from disk instead of memory
+- Simple `echo $content` operations take excessive time
+- Response time increases by 10-100x
+
+**Symptoms of buffer issues:**
+
+- Slow response times for pages with large HTML (>50KB)
+- High disk I/O during request processing
+- PHP execution time is fast, but total response time is slow
+- Performance degrades under load as disk becomes bottleneck
+
+### Buffer configuration with docker-sdk
+
+{% info_block infoBox "Project-level nginx configuration" %}
+
+nginx configuration is managed by [docker-sdk](https://github.com/spryker/docker-sdk) and cannot be directly edited at the project level. Configuration is controlled through deploy files (`deploy.yml`, `deploy.dev.yml`) with options exposed by docker-sdk.
+
+Buffer optimizations described below are already implemented in recent docker-sdk versions.
+
+{% endinfo_block %}
+
+**Optimized buffer settings (implemented in docker-sdk):**
+
+Recent versions of docker-sdk include optimized FastCGI buffer configuration:
+
+- **fastcgi_buffers**: `16 16k` (256KB total) - [implementation](https://github.com/spryker/docker-sdk/pull/547/files#diff-48456f25a9965f118a75a3f431a2df9e212f6ad5ff35c4ee452692c02c74feb1R498)
+- **fastcgi_buffer_size**: `16k` - [implementation](https://github.com/spryker/docker-sdk/pull/547/files#diff-48456f25a9965f118a75a3f431a2df9e212f6ad5ff35c4ee452692c02c74feb1R502)
+- **fastcgi_max_temp_file_size**: `1m` - [implementation](https://github.com/spryker/docker-sdk/pull/547/files#diff-48456f25a9965f118a75a3f431a2df9e212f6ad5ff35c4ee452692c02c74feb1R519)
```

Comment:
this one links to "`groups: applications: application: endpoints: redirects:` - defines http redirects for the specific configuration." Not sure this is correct.